example/mqtt/test_mqtt.tavern.yaml

```diff
@@ -0,0 +1,112 @@
+---
+
+test_name: Test mqtt message response
+
+includes:
+  - !include common.yaml
+
+mqtt: &mqtt_spec
+  # tls:
+  #   enable: true
+  connect:
+    host: localhost
+    port: 9001
+    timeout: 3
+  client:
+    transport: websockets
+    client_id: tavern-tester
+
+stages:
+  - name: step 1 - ping/pong
+    mqtt_publish:
+      topic: /device/123/ping
+      payload: ping
+    mqtt_response:
+      topic: /device/123/pong
+      payload: pong
+      timeout: 5
+
+---
+
+test_name: Make sure posting sends mqtt message
+
+includes:
+  - !include common.yaml
+
+mqtt: *mqtt_spec
+
+stages:
+  - name: step 1 - post message trigger
+    request:
+      url: "{host}/send_mqtt_message"
+      json:
+        device_id: 123
+        payload: "hello"
+      method: POST
+      headers:
+        content-type: application/json
+    response:
+      status_code: 200
+      body:
+        topic: "/device/123"
+      headers:
+        content-type: application/json
+    mqtt_response:
+      topic: /device/123
+      payload: "hello"
```

Comment:
How might we validate JSON or other encoded payloads?

Reply:
Now done using the same method as validating json responses from a server. We can rework that function (`BaseResponse.recurse_check_key_match`) to make this better in future (eg https://github.com/taverntesting/tavern/issues/20)